services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/oauth/OpenIDConnectManagerImplUnitTest.java

```diff
@@ -808,18 +898,217 @@ public void testGetAccessToken_noOpenIdScope() {
+				openIDConnectManagerImpl.generateTokenResponseWithRefreshToken(refreshToken, OAUTH_CLIENT_ID, "offline_access", OAUTH_ENDPOINT));
+	}
+
+	@Test
+	public void testGetTokenResponseWithRefreshToken_exceedingScope() {
+		when(mockOauthClientDao.isOauthClientVerified(OAUTH_CLIENT_ID)).thenReturn(true);
+
+		// This will be the new token and metadata
+		OAuthRefreshTokenAndMetadata expectedRefreshTokenAndId = createRotatedToken();
+
+		String refreshToken = "pre-generated refresh token";
+		when(oauthRefreshTokenManager.rotateRefreshToken(refreshToken)).thenReturn(expectedRefreshTokenAndId);
+
+		String scope = "openid offline_access authorize"; // Authorize was not previously granted
+
+		// method under test
+		assertThrows(IllegalArgumentException.class, () -> openIDConnectManagerImpl.generateTokenResponseWithRefreshToken(refreshToken, OAUTH_CLIENT_ID, scope, OAUTH_ENDPOINT));
+	}
+
+	@Test
+	public void testGetTokenResponseWithRefreshToken_noOpenIdScope() {
+		when(mockOauthClientDao.isOauthClientVerified(OAUTH_CLIENT_ID)).thenReturn(true);
+		when(mockOauthClientDao.getSectorIdentifierSecretForClient(OAUTH_CLIENT_ID)).thenReturn(clientSpecificEncodingSecret);
+		when(mockClock.currentTimeMillis()).thenReturn(System.currentTimeMillis());
+
+		// This will be the new token and metadata
+		OAuthRefreshTokenAndMetadata expectedRefreshTokenAndId = createRotatedToken();
+		Date initalAuthzOn = new Date(System.currentTimeMillis() - 1000 * 60 * 60 * 24);
+		expectedRefreshTokenAndId.getMetadata().setAuthorizedOn(initalAuthzOn);
+
+
+		String refreshToken = "pre-generated refresh token";
+		when(oauthRefreshTokenManager.rotateRefreshToken(refreshToken)).thenReturn(expectedRefreshTokenAndId);
+
+		String expectedAccessToken = "ACCESS-TOKEN";
+		when(oidcTokenHelper.createOIDCaccessToken(eq(OAUTH_ENDPOINT), eq(ppid), eq(OAUTH_CLIENT_ID), anyLong(),
+				eq(initalAuthzOn), eq(expectedRefreshTokenAndId.getMetadata().getTokenId()), anyString(), scopesCaptor.capture(), claimsCaptor.capture())).thenReturn(expectedAccessToken);
+
+		String scope = "offline_access"; // Do not request openid!
+		// elsewhere we test that we correctly build up the requested user-info
+		// here we just spot check a few fields to make sure everything's wired up
+
+		// method under test
+		OIDCTokenResponse tokenResponse = openIDConnectManagerImpl.generateTokenResponseWithRefreshToken(refreshToken, OAUTH_CLIENT_ID, scope, OAUTH_ENDPOINT);
+
+		// verifying the mock token indirectly verifies all param's were correctly passed to oidcTokenHelper.createOIDCIdToken()
+		assertNull(tokenResponse.getId_token());
+
+		assertEquals(expectedAccessToken, tokenResponse.getAccess_token());
+		assertEquals(Collections.singletonList(OAuthScope.offline_access), scopesCaptor.getValue());
+		assertEquals(expectedRefreshTokenAndId.getRefreshToken(), tokenResponse.getRefresh_token());
+	}
+
+	@Test
+	public void testGetTokenResponseWithRefreshToken_nullOrEmptyScope() {
```

Comment:
please remove claims as a request parameter to OpenIDController.getTokenResponse after double checking that the spec' does not require it.